pubsub/gcp/pubsub/pubsub.go

```diff
@@ -76,9 +82,37 @@ type WhatNow struct {
 	Type string `json:"type"`
 }
 
+const topicCacheRefreshInterval = 5 * time.Hour
+
 // NewGCPPubSub returns a new GCPPubSub instance.
 func NewGCPPubSub(logger logger.Logger) pubsub.PubSub {
-	return &GCPPubSub{logger: logger, closeCh: make(chan struct{})}
+	client := &GCPPubSub{
+		logger:     logger,
+		closeCh:    make(chan struct{}),
+		topicCache: make(map[string]cacheEntry),
+		lock:       &sync.RWMutex{},
+	}
+	return client
+}
+
+func (g *GCPPubSub) periodicCacheRefresh() {
+	ticker := time.NewTicker(topicCacheRefreshInterval / 5)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-g.closeCh:
+			return
+		case <-ticker.C:
+			g.lock.Lock()
+			for key, entry := range g.topicCache {
+				if time.Since(entry.LastSync) > topicCacheRefreshInterval/5 {
```

Comment:
```suggestion
                // Delete from the cache if the last sync was longer than topicCacheRefreshInterval
				if time.Since(entry.LastSync) > topicCacheRefreshInterval {
```